tests/unit/test_kmod_averaging.py

```diff
@@ -0,0 +1,83 @@
+from pathlib import Path
+
+import pandas.testing as pdt
+import pytest
+import tfs
+
+from omc3.scripts.kmod_average import (
+    AVERAGED_BETASTAR_FILENAME,
+    AVERAGED_BPM_FILENAME,
+    EXT,
+    average_kmod_results,
+)
+from omc3.plotting.plot_kmod_results import PARAM_BETA, PARAM_WAIST
+from tests.conftest import INPUTS, ids_str
+
+KMOD_INPUT_DIR = INPUTS / "kmod"
+REFERENCE_DIR = KMOD_INPUT_DIR / "references"
+
+@pytest.mark.basic
+@pytest.mark.parametrize("ip", [1, 5], ids=ids_str("ip{}"))
+@pytest.mark.parametrize("n_files", [1, 2], ids=ids_str("{}files"))
+def test_kmod_averaging(tmp_path, ip, n_files):
+    beta = 0.22
+    meas_paths = [_get_measurement_dir(ip, i+1) for i in range(n_files)]
+    ref_output_dir = _get_reference_dir(ip, n_files)
+
+    average_kmod_results(
+        meas_paths=meas_paths, 
+        output_dir=tmp_path,
+        ip=ip, 
+        betastar=[beta],
+        plot=True
+     )
+    _assert_correct_files_are_present(tmp_path, ip, beta)
+
+    for out_name in _get_all_tfs_filenames(ip, beta):
+        out_file = tfs.read(tmp_path / out_name)
+        ref_file = tfs.read(ref_output_dir / out_name)
+        pdt.assert_frame_equal(out_file, ref_file, check_like=True)
```

Comment:
Can we also check for the headers too? I think we should, some of the kmod scripts info is stored there.